Source/WebCore/platform/graphics/adwaita/Adwaita.h

```diff
@@ -0,0 +1,137 @@
+constexpr auto sliderThumbBackgroundHoveredColorDark = SRGBA<uint8_t> { 230, 230, 230 };
+constexpr auto sliderThumbBackgroundDisabledColorDark = SRGBA<uint8_t> { 150, 150, 150 };
+
+constexpr int buttonFocusOffset = -2;
+constexpr int buttonBorderSize = 1;
+
+constexpr auto buttonTextColorLight = SRGBA<uint8_t> { 0, 0, 0, 204 };
+constexpr auto buttonTextDisabledColorLight = SRGBA<uint8_t> { 0, 0, 0, 102 };
+constexpr auto buttonTextColorDark = SRGBA<uint8_t> { 255, 255, 255 };
+constexpr auto buttonTextDisabledColorDark = SRGBA<uint8_t> { 255, 255, 255, 127 };
+
+constexpr auto buttonBorderColorLight = SRGBA<uint8_t> { 0, 0, 0, 50 };
+constexpr auto buttonBackgroundColorLight = SRGBA<uint8_t> { 244, 244, 244 };
+constexpr auto buttonBackgroundPressedColorLight = SRGBA<uint8_t> { 214, 214, 214 };
+constexpr auto buttonBackgroundHoveredColorLight = SRGBA<uint8_t> { 248, 248, 248 };
+constexpr auto toggleBorderColorLight = SRGBA<uint8_t> { 0, 0, 0, 50 };
+constexpr auto toggleBorderHoveredColorLight = SRGBA<uint8_t> { 0, 0, 0, 80 };
+
+constexpr auto buttonBorderColorDark = SRGBA<uint8_t> { 255, 255, 255, 50 };
+constexpr auto buttonBackgroundColorDark = SRGBA<uint8_t> { 52, 52, 52 };
+constexpr auto buttonBackgroundPressedColorDark = SRGBA<uint8_t> { 30, 30, 30 };
+constexpr auto buttonBackgroundHoveredColorDark = SRGBA<uint8_t> { 60, 60, 60 };
+constexpr auto toggleBorderColorDark = SRGBA<uint8_t> { 255, 255, 255, 50 };
+constexpr auto toggleBorderHoveredColorDark = SRGBA<uint8_t> { 255, 255, 255, 80 };
+
+constexpr double toggleSize = 14.;
+constexpr int toggleBorderSize = 2;
+constexpr int toggleFocusOffset = 1;
+
+constexpr auto spinButtonBorderColorLight = SRGBA<uint8_t> { 0, 0, 0, 25 };
+constexpr auto spinButtonBackgroundColorLight = Color::white;
+constexpr auto spinButtonBackgroundHoveredColorLight = SRGBA<uint8_t> { 0, 0, 0, 50 };
+constexpr auto spinButtonBackgroundPressedColorLight = SRGBA<uint8_t> { 0, 0, 0, 70 };
+
+constexpr auto spinButtonBorderColorDark = SRGBA<uint8_t> { 255, 255, 255, 25 };
+constexpr auto spinButtonBackgroundColorDark = SRGBA<uint8_t> { 45, 45, 45 };
+constexpr auto spinButtonBackgroundHoveredColorDark = SRGBA<uint8_t> { 255, 255, 255, 50 };
+constexpr auto spinButtonBackgroundPressedColorDark = SRGBA<uint8_t> { 255, 255, 255, 70 };
+
+
+enum class PaintRounded : bool { No, Yes };
+void paintFocus(GraphicsContext&, const FloatRect&, int offset, const Color&, PaintRounded = PaintRounded::No);
+void paintFocus(GraphicsContext&, const Path&, const Color&);
+void paintFocus(GraphicsContext&, const Vector<FloatRect>&, const Color&, PaintRounded = PaintRounded::No);
+
+enum class ArrowDirection { Up, Down };
+void paintArrow(GraphicsContext&, const FloatRect&, ArrowDirection, bool useDarkAppearance);
+
+inline Color focusColor(const Color& accentColor)
+{
+    return accentColor.colorWithAlphaMultipliedBy(focusRingOpacity);
+}
+
+Color systemAccentColor();
+Color systemFocusRingColor();
```

Comment:
I guess these global methods could be part of the base ControlAdwaita, no?

Reply:
Yes, it could. But, `paintFocus` is called from some classes `GraphicsContextCairo`, `GraphicsContextSkia` and `NicosiaCairoOperationRecorder`.
These classes have to include `ControlAdwaita.h` and `ControlAdwaita::paintFocus()`.
I though including `Adwaita.h` and `Adwaita::paintFocus()` are more concis. But, not strong option. I will change if you prefer.